### ISSUE TITLE
Questions: block-finalisation state transition function:  Ω definition

### ISSUE THREAD

**Issue by WhisperingChaos** [ethereum/yellowpaper#399]
I've some questions regarding my understanding of the paper's notation and wasn't sure where to post them.  Therefore, let me know if there's a better forum to direct my questions.

The paper begins with the following:
(1) **σ<sub>t+1</sub> ≡ Υ(σ<sub>t</sub>, T)**
where **σ<sub>t</sub>** represents the current canonical state (version) of the Ethereum World and **T** represents a transaction to be applied to this state.  Once applied by **Y**, **T** yields a new World state.

Soon after defining the above, the paper describes block-finalization by
(2) **σ<sub>t+1</sub> ≡ Π(σ<sub>t</sub>, B)**
Since **B** is substituted for **T**, the above represents **B** as a Transaction.
(3) **B ≡ (...,(T<sub>0</sub>, T<sub>1</sub>, ...))**
This defines **B** as containing components some being transactions: **(T<sub>0</sub>, T<sub>1</sub>, ...)** while other components **"...,"** aren't defined.  I believe this definition casts **B** as an aggregate Transaction.

(4) **Π(σ, B) ≡ Ω(B, Υ(Υ(σ, T<sub>0</sub>), T<sub>1</sub>)...)**
In this equation, **B** on the right hand side becomes **σ<sub>t</sub>** suggesting **B** represents the current canonical state (version) of the Ethereum World.  While its aggregated transactions **Υ(Υ(σ, T<sub>0</sub>), T<sub>1</sub>)...** are substituted for **T**.

So here are my questions:

- In (4) above, **B** presents a duality:  it seems both an aggregate Transaction and the current canonical state (version) **σ<sub>t</sub>** of Ethereum.  Is this true?
- When applying  **Υ(Υ(σ, T<sub>0</sub>), T<sub>1</sub>)...)** the answer is **σ<sub>t</sub>** where **<sub>t</sub>** represents number of transactions in B. Therefore, rewriting (4) generates:
 **Π(σ, B) ≡ Ω(B, σ<sub>t</sub>)**
 Does (4) violate (1) where **B** should represent **σ<sub>t</sub>** and **σ<sub>t</sub>** should represent a Transaction?

Perhaps (4) could be written as:

**(4) Π(σ<sub>t</sub>, B) ≡ Ω( Y<sub>n</sub>(... (Υ<sub>1</sub></sub>(Υ<sub>0</sub>(σ<sub>t=0</sub>, T<sub>t=0</sub>), T<sub>t=1</sub>) ,... ) ,T<sub>t=n</sub>) ,R)**

Where **R** represents the rewards Transaction issued to a nominated party.

Thanks for any insight you can offer.

**Comment by chronaeon** [comment]
Great points, glad to see someone else brought this up. Similar to my remarks in: https://github.com/ethereum/yellowpaper/issues/457.

**Comment by jamesray1** [comment]
Yes, a block aggregates transactions, represents the state of Ethereum 1.0 at a point in time, like a message in a ledger, where this message is then gossiped amongst validators and miners (in a peer to peer network) until it is accepted into the GHOST, the greediest, heaviest observed subtree, or rejected if not.

**Comment by jamesray1** [comment]
For the second question, I don't think there is a violation, but it isn't very clear, and it could be rewritten as you propose. Would you like to make a pull request?
